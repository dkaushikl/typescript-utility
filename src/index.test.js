import {expect} from 'chai'
import names from '.'

describe('names', () => {
  it('should have a list of all available names', () => {
    expect(names.all).to.satisfy(isArrayOfStrings)
  })

  it('should allow me to get a random name from the list', () => {
    expect(names.random()).to.satisfy(isIncludedIn(names.all))
  })
})

function isArrayOfStrings(array) {
  return array.every(i => typeof i === 'string')
}

function isIncludedIn(array) {
  return item => array.includes(item)
}
