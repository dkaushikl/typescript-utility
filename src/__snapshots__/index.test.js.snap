// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`all function should be available > should have a list of all available names 1`] = `true`;
